### DISCUSSION TITLE
AG Plugin 1.2.0 (BETA 1) race condition: port 55056 can't bind (Windows)

### DISCUSSION THREAD

**Discussion by jaacMusic** [apohl79/audiogridder#504] [Help]
It seems when the AG server is restarted there is a race condition where port 55056 gets locked/used when restarting AG server. It would say the port is already in use. I found out closing all AG server instances from the task manager and re-running the AG server fixes the port issue.

Another item is that it seems the AG plugin connects to the AG server by referencing it by IP not by hostname/instance name. This is an issue because when I change from the main AG server to a backup AG server and I have to go to every AG plugin instance in the project and select the new entry by IP. Also, it would be nice if an AG plugin instance is changed to point to a different server to ask if I would like to assign all other instances to this AG server.

**Comment by Fittkenny** [comment]
I've seen this also....

**Comment by apohl79** [comment]
> It seems when the AG server is restarted there is a race condition where port 55056 gets locked/used when restarting AG server. It would say the port is already in use. I found out closing all AG server instances from the task manager and re-running the AG server fixes the port issue.

This can happen with sandboxing on. If a sandbox is still running when the server restarts and comes back up, it might still block the port. I was too "lazy" to handle this properly so far, so this should work properly in the future.

> Another item is that it seems the AG plugin connects to the AG server by referencing it by IP not by hostname/instance name. This is an issue because when I change from the main AG server to a backup AG server and I have to go to every AG plugin instance in the project and select the new entry by IP. Also, it would be nice if an AG plugin instance is changed to point to a different server to ask if I would like to assign all other instances to this AG server.

Hostname is as unreliable as IP, I find IP more reliable in my case tbh, but yes a remote name/address can always change. Instance name is not backed up by the network, so it would only work when you are on the same network and it would slow down session loading, as the plugins have to figure out active servers first.

Long story short, what you are looking for is in the new tray app (the main purpose why I added it btw). You can reconnect all plugins to a new server at once.
